mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoDeleteMethodGenerator.java

```diff
@@ -70,12 +72,9 @@ public DaoDeleteMethodGenerator(
     Delete annotation = methodElement.getAnnotation(Delete.class);
     assert annotation != null;
     if (annotation.ifExists() && !annotation.customIfClause().isEmpty()) {
-      context
-          .getMessager()
-          .error(
-              methodElement,
-              "Invalid annotation parameters: %s cannot have both ifExists and customIfClause",
-              Delete.class.getSimpleName());
+      methodMessager.error(
```

Comment:
Possibly related to my previous comment: wouldn't it be possible to obtain the messager from the context in all cases, and let the context decide which specific implementation to use?